atmo/jobs/models.py

```diff
@@ -98,20 +100,50 @@ def __str__(self):
     def __repr__(self):
         return "<SparkJob {} with {} nodes>".format(self.identifier, self.size)
 
+    @property
+    def has_never_run(self):
+        """
+        Whether the job has run before.
+        Looks at both the cluster status and our own record when
+        we asked it to run.
+        """
+        return (self.most_recent_status == self.DEFAULT_STATUS and
+                self.last_run_date is None)
+
+    @property
+    def has_finished(self):
+        """Whether the job's cluster is terminated or failed"""
+        return self.most_recent_status in self.FINAL_STATUS_LIST
+
+    @property
+    def is_runnable(self):
+        return self.has_never_run or self.has_finished
+
+    @property
+    def is_expired(self):
+        """Whether the current job run has run out of time"""
+        if not self.current_run_jobflow_id or self.last_run_date is None:
```

Comment:
@maurodoglio Here's still a chance of a race condition since we check `current_run_jobflow_id` and `last_run_date` to decide if a job is expired and not `most_recent_status` to decide if it's expired or not.

The first are stored by atmo when a job is run, the latter when we update the status from AWS. Which means I'm wondering if that's the right condition.

In the tests there is this:
```
    # A spark job cannot expire if:
    # it doesn't have a jobflow_id OR
    # it doesn't have a a last_run_date OR
    # its most_recent_status is not RUNNING OR
    # it hasn't run for more than its timeout
```